src/Plugin/Field/FieldFormatter/StrawberryMediaFormatter.php

```diff
@@ -291,7 +294,8 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
                   // This also never runs if cached. So after deletion we better
                   // call the controller!
                   if (!empty($jsondata['ap:annotationCollection']) && is_array($jsondata['ap:annotationCollection'])) {
-                    WebAnnotationController::primeKeyStore($items[$delta]);
+                    $keystoreid = $elements[$delta]['media' . $i]['#attached']['drupalSettings']['format_strawberryfield']['openseadragon'][$uniqueid]['keystoreid'];
+                    WebAnnotationController::primeKeyStore($keystoreid);
```

Comment:
Hi Pedro, sorry again. I explained myself wrong here: It is both, look at the primeKeyStore signature. 
```PHP
/**
   * Primes the Web Annotation KeyStore with saved values.
   *
   * @param \Drupal\strawberryfield\Plugin\Field\FieldType\StrawberryFieldItem $itemfield
   * @param null $keystoreid
   */
  public static function primeKeyStore(StrawberryFieldItem $itemfield,  $keystoreid = NULL) {
    if ($keystoreid == NULL && strlen(trim($keystoreid)) == 0) {
      return NULL;
    }
    $jsondata = $itemfield->provideDecoded(TRUE);
    $tempstore = \Drupal::service('tempstore.private')->get(
      'webannotation'
    );
    if (!empty($jsondata['ap:annotationCollection']) && is_array($jsondata['ap:annotationCollection'])) {
      $tempstore->set($keystoreid, $jsondata['ap:annotationCollection']);
      return $jsondata['ap:annotationCollection'];
    }
    else {
      $tempstore->set($keystoreid, []);
      return [];
    }
  }
``` this won't work that way. You can look at how we call this in other parts.

The issue I had was that I was passing only the $items[$delta] when we already know where we want the keystore to be primed. But also this needs some little testing if you could. Thanks

Reply:
Ah I see that now, not sure why PHPStorm didn't call me out when I did that